common-app-lib/src/cmr/common_app/data/search/collection_for_gran_acls_by_concept_id_cache.clj

```diff
@@ -0,0 +1,55 @@
+(ns cmr.common-app.data.search.collection-for-gran-acls-by-concept-id-cache
+"Defines common functions and defs a cache for search collections.
+ The collections cache contains data like the following:
+
+ concept-id -> {collection info}"
+
+(:require
+  [cmr.redis-utils.redis-hash-cache :as red-hash-cache]
+  [cmr.common.hash-cache :as hash-cache]
+  [cmr.common.jobs :refer [defjob]]
+  [cmr.common.log :as log :refer (debug info warn error)]
+  [cmr.common.services.errors :as errors]))
+
+(def cache-key
+  "Identifies the key used when the cache is stored in the system."
+  :collections-for-gran-acls-by-concept-id)
+
+(def initial-cache-state
+  "The initial cache state."
+  nil)
+
+(def job-refresh-rate
+  "This is the frequency that the cron job will run. It should be less than coll-for-gran-acl-by-concept-id-ttl"
+  ;; 15 minutes
+  (* 60 15))
+
+(def cache-ttl
+  "This is when Redis should expire the data, this value should never be hit if
+   the cron job is working correctly, however in some modes (such as local dev
+   with no database) it may come into play."
+   ;; 30 minutes
+  (* 60 30))
+
+(defn create-cache
+  "Creates a new empty collections cache."
+  []
+  (red-hash-cache/create-redis-hash-cache {:keys-to-track [cache-key]
+                                           :ttl           cache-ttl}))
+;; FIXME
+(defn refresh-cache
```

Comment:
Might move to bootstrap depending on how 'get-collections' is going to be refactored.